codegens/nodejs-request/lib/request.js

```diff
@@ -14,9 +14,18 @@ var _ = require('./lodash'),
  */
 function makeSnippet (request, indentString, options) {
   var snippet = 'var request = require(\'request\');\n',
-    optionsArray = [];
-
-  snippet += 'var fs = require(\'fs\');\n';
+    optionsArray = [],
+    isFormDataFile = false;
+  if (request.body && request.body.mode === 'formdata') {
```

Comment:
why do we need .toJSON when reading `formdata` and not while reading `mode`?

Reply:
.mode is available before .toJSON but for reading the formdata array, we convert it to .toJSON for easy access